lib/matplotlib/pyplot.py

```diff
@@ -2283,12 +2283,15 @@ def polar(*args, **kwargs):
     strings, as in :func:`~matplotlib.pyplot.plot`.
 
     """
-    # If an axis already exists, check if it has a polar projection
+    # If an axes already exists, check if it has a polar projection
     if gcf().get_axes():
         if not isinstance(gca(), PolarAxes):
             warnings.warn('Trying to create polar plot on an axis that does '
                           'not have a polar projection.')
-    ax = gca(polar=True)
+        ax = gca()
+    else:
+        ax = gcf().add_subplot(1, 1, 1, polar=True)
```

Comment:
umm, I'm not too up on the pyplot interface, but is `gca()` necessarilly a `subplot(111)`?

Reply:
No, but: If there are axes, use the gca(). If not, make a new one equivalent to subplot(111).

Reply:
OK, but I don't get the point of this PR then.  Why is this better than whats there now?  Is the idea to let `plt.polar` write on an existing polar axes, despite the fact `gca` will create a new axes in the future?  That doesn't seem right.  Sorry if I'm being dense...

Reply:
The whole `gca`situation is incredibly confusing.  I can't make sense of the current behavior and docstrings, and the warning about future behavior makes no sense to me either.  `gca` means "get current axes"; it should not mean "always get a new axes".  I think that the coming change to `subplot` is being incorrectly applied to `gca`.

Reply:
The change was made in https://github.com/matplotlib/matplotlib/pull/10660 - I think `gca` should just get the current axis, and not create a new one, and that change should be reverted. ping @anntzer

Reply:
I agree that the warning in #10660 is wrong.  I need to rethink about the correct way to ultimately get rid of the axes collision behavior.

Reply:
Great, I'll close this then.